sensor/common/scannerdefinitions/http_handler.go

```diff
@@ -20,29 +22,48 @@ var (
 	headersToProxy = set.NewFrozenStringSet("If-Modified-Since", "Accept-Encoding")
 )
 
-// scannerDefinitionsHandler handles requests to retrieve scanner definitions
+// ScannerDefinitionHandler handles requests to retrieve scanner definitions
 // from Central.
-type scannerDefinitionsHandler struct {
-	centralClient *http.Client
+type ScannerDefinitionHandler struct {
+	centralClient    *http.Client
+	centralReachable *atomic.Bool
 }
 
 // NewDefinitionsHandler creates a new scanner definitions handler.
-func NewDefinitionsHandler(centralEndpoint string) (http.Handler, error) {
+func NewDefinitionsHandler(centralEndpoint string) (*ScannerDefinitionHandler, error) {
 	client, err := clientconn.NewHTTPClient(mtls.CentralSubject, centralEndpoint, 0)
 	if err != nil {
 		return nil, errors.Wrap(err, "instantiating central HTTP transport")
 	}
-	return &scannerDefinitionsHandler{
-		centralClient: client,
+	return &ScannerDefinitionHandler{
+		centralClient:    client,
+		centralReachable: &atomic.Bool{},
 	}, nil
 }
 
-func (h *scannerDefinitionsHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+// Notify reacts to sensor going into online/offline mode.
+func (h *ScannerDefinitionHandler) Notify(e common.SensorComponentEvent) {
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		h.centralReachable.Store(true)
+	case common.SensorComponentEventOfflineMode:
+		h.centralReachable.Store(false)
+	}
+}
+
+func (h *ScannerDefinitionHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 	// Validate request.
 	if request.Method != http.MethodGet {
 		writer.WriteHeader(http.StatusMethodNotAllowed)
 		return
 	}
+
+	// If central is not reachable, then the request should return an error to Scanner.
+	if !h.centralReachable.Load() {
+		httputil.WriteGRPCStyleErrorf(writer, codes.Internal, "central not reachable")
```

Comment:
How about using the Unavailable code?

```
	// Unavailable indicates the service is currently unavailable.
	// This is a most likely a transient condition and may be corrected
	// by retrying with a backoff. Note that it is not always safe to retry
	// non-idempotent operations.
	//
	// See litmus test above for deciding between FailedPrecondition,
	// Aborted, and Unavailable.
	//
	// This error code will be generated by the gRPC framework during
	// abrupt shutdown of a server process or network connection.
```
from: https://pkg.go.dev/google.golang.org/grpc/codes

```suggestion
		httputil.WriteGRPCStyleErrorf(writer, codes.Unavailable, "central not reachable")
```

Reply:
Done, thanks!